ext/date/php_date.c

```diff
@@ -3699,7 +3700,8 @@ PHP_METHOD(DateTimeZone, __set_state)
 	php_date_instantiate(date_ce_timezone, return_value);
 	tzobj = Z_PHPTIMEZONE_P(return_value);
 	if(php_date_timezone_initialize_from_hash(&return_value, &tzobj, myht) != SUCCESS) {
-		php_error_docref(NULL, E_ERROR, "Timezone initialization failed");
+		zend_throw_error(zend_ce_error, "Timezone initialization failed");
+		RETURN_FALSE;
```

Comment:
`return_value` was initialized to a `date_ce_timezone` object above, so this likely needs to be dtor'd here.